tharsis/resource_managed_identity_access_rules.go

```diff
@@ -0,0 +1,328 @@
+package tharsis
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	tharsis "gitlab.com/infor-cloud/martian-cloud/tharsis/tharsis-sdk-go/pkg"
+	ttypes "gitlab.com/infor-cloud/martian-cloud/tharsis/tharsis-sdk-go/pkg/types"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces
+var (
+	_ resource.Resource                = &managedIdentityAccessRuleResource{}
+	_ resource.ResourceWithConfigure   = &managedIdentityAccessRuleResource{}
+	_ resource.ResourceWithImportState = &managedIdentityAccessRuleResource{}
+)
+
+// NewManagedIdentityAccessRuleResource is a helper function to simplify the provider implementation.
+func NewManagedIdentityAccessRuleResource() resource.Resource {
+	return &managedIdentityAccessRuleResource{}
+}
+
+type managedIdentityAccessRuleResource struct {
+	client *tharsis.Client
+}
+
+// Metadata returns the full name of the resource, including prefix, underscore, instance name.
+func (t *managedIdentityAccessRuleResource) Metadata(ctx context.Context,
+	req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "tharsis_managed_identity_access_rule"
+}
+
+// The diagnostics return value is required by the interface even though this function returns only nil.
+func (t *managedIdentityAccessRuleResource) GetSchema(_ context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	description := "Defines and manages a managed identity access rule."
+
+	return tfsdk.Schema{
+		Version: 1,
+
+		MarkdownDescription: description,
+		Description:         description,
+
+		Attributes: map[string]tfsdk.Attribute{
+			"id": {
+				Type:                types.StringType,
+				MarkdownDescription: "String identifier of the access rule.",
+				Description:         "String identifier of the access rule.",
+				Computed:            true,
+			},
+			"run_stage": {
+				Type:                types.StringType,
+				MarkdownDescription: "Type of job, plan or apply.",
+				Description:         "Type of job, plan or apply.",
+				Required:            true,
+			},
+			"managed_identity_id": {
+				Type:                types.StringType,
+				MarkdownDescription: "String identifier of the connected managed identity.",
+				Description:         "String identifier of the connected managed identity.",
+				Required:            true,
+			},
+			"allowed_users": {
+				Type: types.SetType{
+					ElemType: types.StringType,
+				},
+				MarkdownDescription: "List of email addresses of users allowed to use this rule.",
```

Comment:
These descriptions can be updated since the allowed principals doesn't control who can use this rule but rather who can use the managed identity associated with the rule for the specified run statge.